### PR TITLE
Highlighting functionality to determine best matching version of address

### PR DESCRIPTION
**Pull request by RichardSmithONS**

### CODE REVIEW

File: model/src/main/scala/uk/gov/ons/addressIndex/model/db/index/HybridAddress.scala [hunk 1]
```diff
@@ -87,6 +89,8 @@ object HybridAddress {
                             else if (eWDistance > niDistance) niDistance
                               else (eWDistance + centimetre)
 
+      val highlights = hit.asInstanceOf[SearchHit].highlight
+  //    println("highlights = " + highlights)
```
Comment (steve-thorne-ONS) [inline model/src/main/scala/uk/gov/ons/addressIndex/model/db/index/HybridAddress.scala]: Should remove this.

File: model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseAddress.scala [hunk 2]
```diff
@@ -33,9 +34,11 @@ case class AddressResponseAddress(uprn: String,
                                   censusEstabType: String,
                                   countryCode: String,
                                   lpiLogicalStatus: String,
-                                  fromSource: String,
+                              //    fromSource: String,
```
Comment (steve-thorne-ONS) [inline model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseAddress.scala]: Remove?

File: model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseAddress.scala [hunk 3]
```diff
@@ -33,9 +34,11 @@ case class AddressResponseAddress(uprn: String,
...
                                   confidenceScore: Double,
-                                  underlyingScore: Float)
+                                  underlyingScore: Float,
+                               //   bestMatchField: String
```
Comment (steve-thorne-ONS) [inline model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseAddress.scala]: Remove?

File: model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseAddress.scala [hunk 4]
```diff
@@ -96,12 +102,13 @@ object AddressResponseAddress {
       censusEstabType = other.censusEstabType,
       countryCode = other.countryCode,
       lpiLogicalStatus = lpiLogicalStatus,
-      fromSource = other.fromSource,
       confidenceScore = 100D,
-      underlyingScore = if (other.distance == 0) other.score else (other.distance/1000).toFloat
+      underlyingScore = if (other.distance == 0) other.score else (other.distance/1000).toFloat,
```
Comment (steve-thorne-ONS) [inline model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseAddress.scala]: What's the , for?

Comment (RichardSmithONS) [reply model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseAddress.scala]: Scala 2.12 lets you have comma on the last item. I think there was another field after the underlying score at one point.

File: model/src/main/scala/uk/gov/ons/addressIndex/model/db/index/HybridAddress.scala (hunk 1 above)
Comment (RichardSmithONS) [reply model/src/main/scala/uk/gov/ons/addressIndex/model/db/index/HybridAddress.scala]: ok

File: model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseAddress.scala (hunk 2 above)
Comment (RichardSmithONS) [reply model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseAddress.scala]: ok

File: model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseAddress.scala (hunk 3 above)
Comment (RichardSmithONS) [reply model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseAddress.scala]: ok

File: server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala [hunk 5]
```diff
@@ -155,6 +156,8 @@ class AddressControllerSpec extends PlaySpec with Results {
     source = "7666OW"
   )
 
+  // todo add test highlighting
```
Comment (steve-thorne-ONS) [inline server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala]: When?

Comment (RichardSmithONS) [reply server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala]: I was intending to leave it empty for now (whilst we're in iterative mode), and as the highlighting only applies to partials it's likely to cause quite a bit of trouble / failed tests.

**Comment by RichardSmithONS** [comment]
> I had started to create a EQ specific /partial endpoint that returned limited data (uprn and address) looks like I'll have to merge the highlighting into it.

Yes. As discussed in the recent email discuusion the EQ endpoint will indeed need to do stuff with the highlights so you'll need to merge in the code after the PR.

**Pull request merged**